### PR TITLE
Add missing call to "VolumeDriver.Mount" during container start.

### PR DESCRIPTION
**Pull request by anusha-ragunathan**
On container start, volume mounts are setup which should include
invoking "VolumeDriver.Mount" to setup necessary mounts. This was
missing for Windows containers. Fix that.

Signed-off-by: Anusha Ragunathan anusha@docker.com

### CODE REVIEW

**Review by cpuguy83 — APPROVED** [review]
LGTM if janky is happy

**Comment by thaJeztah** [comment]
ping @jhowardmsft PTAL

**Comment by jhowardmsft** [comment]
I'm wondering if this adversely affects Hyper-V containers? @swernli

**Comment by anusha-ragunathan** [comment]
win2lin is failing due to unrelated known issue 26506

**Comment by jhowardmsft** [comment]
I'll verify this against Hyper-V containers this afternoon, unless you already have @anusha-ragunathan. Myself and @swernli did a quick code inspection, but weren't able to tell if it would regress them. CI only uses Windows Server containers.

**Comment by jhowardmsft** [comment]
@anusha-ragunathan 

I did the following manual verification against this branch. Anything else I should be verifying? I'm not entirely sure what scenarios this changes addresses (sorry, not had time to look in detail)

```docker run -it --rm -v c:\packer:c:\packer --isolation=hyperv nanoserver cmd /s /c dir c:\packer```

```
 Volume in drive C has no label.
 Volume Serial Number is 8C12-2CCC

 Directory of c:\packer

09/13/2016  03:21 PM    <DIR>          .
09/13/2016  03:21 PM    <DIR>          ..
09/13/2016  03:11 PM             2,782 authorized_keys
07/22/2016  02:27 PM             7,488 Bootstrap.ps1
09/13/2016  03:21 PM            16,768 configure.log
09/13/2016  03:11 PM               742 ConfigureCIEnvironment.ps1
09/13/2016  03:11 PM             2,762 ConfigureControlDaemon.ps1
09/13/2016  03:11 PM             1,420 ConfigureSSH.ps1
09/13/2016  03:11 PM               697 ConfigureSSH.sh
09/13/2016  03:11 PM             2,640 DownloadOSImages.ps1
09/13/2016  03:12 PM             6,728 DownloadScripts.ps1
09/13/2016  03:11 PM            16,711 InstallMostThings.ps1
09/13/2016  03:11 PM             1,002 nssmdocker.cmd
09/13/2016  03:11 PM             5,877 Phase0.ps1
09/13/2016  03:11 PM                 0 Phase0.RebootedOnce.txt
09/13/2016  03:11 PM            11,675 Phase1.ps1
09/13/2016  03:11 PM             2,623 Phase2.ps1
09/13/2016  03:11 PM             4,173 Phase3.ps1
09/13/2016  03:11 PM             2,409 Phase4.ps1
09/13/2016  03:11 PM             2,859 Phase5.ps1
              18 File(s)         89,356 bytes
               2 Dir(s)  98,016,514,048 bytes free
PS C:\>
```

**Comment by jhowardmsft** [comment]
I'm also attempting to do a local CI run with this commit using Hyper-V containers.

**Comment by cpuguy83** [comment]
@jhowardmsft Really what this is doing is making sure that `volume.Mount()` is called before trying to stick it in a container, otherwise the volume driver will not have had a chance to mount the data to the host. In many cases `volume.Path()` may be empty until `volume.Mount()` is called.

**Comment by jhowardmsft** [comment]
LGTM. I'm not seeing any failures in the local CI run with Hyper-V containers which relates to anything here (CI doesn't run fully against Hyper-V containers yet).

**Comment by thaJeztah** [comment]
Thanks @jhowardmsft, looks like we have two LGTM's, and all green. Merging

**Pull request merged**
